### PR TITLE
Response decompression: rework batch test to WireMock

### PR DESCRIPTION
**Pull request by vlsi**
Note: this builds on top of https://github.com/apache/jmeter/pull/496

## Motivation and Context

ResponseDecompression test fails too often:
https://travis-ci.org/apache/jmeter/jobs/587884339#L1458
https://travis-ci.org/apache/jmeter/jobs/587201685#L1457
https://travis-ci.org/apache/jmeter/jobs/585431878#L1437
https://travis-ci.org/apache/jmeter/jobs/585146534#L1452
https://travis-ci.org/apache/jmeter/jobs/584778862#L1450
https://travis-ci.org/apache/jmeter/jobs/582277722#L1204

So I refactored it to use WireMock

Unfortunately WireMock does not support `deflate` and `br` compressions, so those are still send requests to the Internet.

### CODE REVIEW

File: src/dist-check/src/test/java/org/apache/jmeter/junit/JMeterTest.java [hunk 1]
```diff
@@ -113,6 +115,20 @@ public JMeterTest(String testName, JMeterGUIComponent gc) {
         guiItem = gc;
     }
 
+    @Override
+    public Description getDescription() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(getName());
+        if (guiItem instanceof TestBeanGUI) {
+            sb.append(" ").append(guiItem.toString());
```
Comment (ham1) [inline src/dist-check/src/test/java/org/apache/jmeter/junit/JMeterTest.java]: Could this repeated `.append(" ")` be added before the `if`s?

**Review by ham1 — COMMENTED** [review]
:+1:

File: src/dist-check/src/test/java/org/apache/jmeter/junit/JMeterTest.java (hunk 1 above)
Comment (vlsi) [reply src/dist-check/src/test/java/org/apache/jmeter/junit/JMeterTest.java]: This commit is not part of this PR. This PR builds on top of https://github.com/apache/jmeter/pull/496, so please comment JUnit5 migration there.

**Comment by pmouawad** [comment]
Hi @vlsi ,
Great idea.
But I think failure mainly comes from Deflate no ?
1/ Why did you remove HR1-DEFLATE sampler ? is it not Deflate ?
2/ My understanding is that it will not fix root cause no ?

But I am ok with PR idea provided HR1-DEFLATE is put back again unless there is a reason (I didn't deeply look still).
Thanks

**Comment by vlsi** [comment]
>But I think failure mainly comes from Deflate no ?

Yes and no. Mainly the failure comes from using the unreliable internet for JMeter testing.
So I would treat "removal / replacement of the URLs" to be workaround rather than a solution.

>1/ Why did you remove HR1-DEFLATE sampler ? is it not Deflate ?

It causes failures. Really. Every time the test fails it was caused by HR1-DEFLATE.

>2/ My understanding is that it will not fix root cause no ?

Removal of the ill site would heal the test.

However I don't think testing deflate is important, since, well, is it really used in practice?

**Comment by pmouawad** [comment]
> > But I think failure mainly comes from Deflate no ?
> 
> Yes and no. Mainly the failure comes from using the unreliable internet for JMeter testing.

Is it network or DDOS protection on server side ?

> So I would treat "removal / replacement of the URLs" to be workaround rather than a solution.
> 
> > 1/ Why did you remove HR1-DEFLATE sampler ? is it not Deflate ?

The idea was to have more than 1 site in test.
> 
> It causes failures. Really. Every time the test failed it was caused by HR1-DEFLATE.
> 
> > 2/ My understanding is that it will not fix root cause no ?
> 
> Removal of the ill site would heal the test.
> 
> However I don't think testing deflate is important, since, well, is it really used in practice?

I think it is. We had bug report, and within our consulting work at UbikLoadPack we frequently see it.

**Comment by vlsi** [comment]
>Is it network or DDOS protection on server side ?

probably

I think brotli and deflate could be reworked to use Docker-based nginx server.

**Comment by pmouawad** [comment]
Could we use this :
https://www.testcontainers.org/

**Pull request merged**